src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/BoundNodeClassWriter.cs

```diff
@@ -842,13 +842,14 @@ private void WriteField(Field field)
                     else if (field.Override)
                     {
                         // We emit a suppression here because the bound nodes use a pattern which is safe, but can't be tracked.
+                        var suppression = FieldNullHandling(node, field.Name) is (NullHandling.Allow or NullHandling.Always) ? "" : "!";
```

Comment:
>FieldNullHandling(node, field.Name) is (NullHandling.Allow or NullHandling.Always) [](start = 42, length = 82)

Could we instead check if type of the field is nullable?

Reply:
> Could we instead check if type of the field is nullable?

We could although `FieldNullHandling()` checks the type and also the 'Null' attribute which seems like a stronger check.

---
In reply to: [603318944](https://github.com/dotnet/roslyn/pull/52203#discussion_r603318944) [](ancestors = 603318944)